packages/test-cases/cases/bind/sanity/output/wasm-as/Module/module.ts

```diff
@@ -0,0 +1,168 @@
+import * as Types from "..";
+
+import {
+  Read,
+  ReadDecoder,
+  Write,
+  WriteSizer,
+  WriteEncoder,
+  Box,
+  BigInt,
+  BigNumber,
+  JSON,
+  Context
+} from "@polywrap/wasm-as";
+
+import {
+  deserializemoduleMethodArgs,
+  serializemoduleMethodResult,
+  deserializeobjectMethodArgs,
+  serializeobjectMethodResult,
+  deserializeoptionalEnvMethodArgs,
+  serializeoptionalEnvMethodResult,
+  deserializeifArgs,
+  serializeifResult
+} from "./serialization";
+
+import {
+  wrap_invoke_args,
+  wrap_invoke_result,
+  wrap_invoke_error,
+  wrap_load_env,
+  InvokeArgs,
+} from "@polywrap/wasm-as";
+
+export abstract class IModule {
+  private _env: Types.Env | undefined;
+
+  abstract moduleMethod(
+    args: Types.Args_moduleMethod
+  ): i32;
+
+  abstract objectMethod(
+    args: Types.Args_objectMethod
+  ): Types.AnotherType | null;
+
+  abstract optionalEnvMethod(
+    args: Types.Args_optionalEnvMethod
+  ): Types.AnotherType | null;
+
+  abstract _if(
+    args: Types.Args__if
+  ): Types._else;
+
+  public get env(): Types.Env {
+    if (this._env === undefined) {
+      throw new Error("Environment is not set, and it is required by this module");
+    }
+    return this._env;
+  }
+
+  public setEnv(env: Types.Env): void {
+    this._env = env;
+  }
+
+  public _wrap_invoke(method_size: u32, args_size: u32, env_size: u32): bool {
```

Comment:
NOTE: the `_wrap_name` patter is reserved for wasm exports, just as `__wrap_name` is reserved for wasm imports.

Reply:
Sure we can rename it to anything, I just keep it the way we did for the `PluginModule`.